### PR TITLE
Group tuple tests into @testsets

### PR DESCRIPTION
**Pull request by HarrisonGrodin**
Solves #18807.

### CODE REVIEW

File: test/tuple.jl [hunk 1]
```diff
@@ -230,39 +283,3 @@ let
...
-    tuple16int = (1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1)
-    @test Tuple16Int(tuple16int) isa Tuple16Int
-end
-
-# PR #21446
-for n = 0:15
-    @test ntuple(identity, Val{n}) == ntuple(identity, n)
-end
-
-# Tuple type ninitialized
```
Comment (tkelman) [inline test/tuple.jl]: looks like you deleted this set without a replacement

Comment (HarrisonGrodin) [inline test/tuple.jl]: @tkelman Fixed. My mistake.

**Comment by kshyatt** [comment]
Welcome to base Julia, @HarrisonGrodin! Thanks for this PR - looks like tests are failing because of some sort of syntax issue. Can you run `make test-tuple` locally and see if you can fix it? Feel free to PM me on Discourse if you get stuck.

**Comment by HarrisonGrodin** [comment]
@kshyatt The issue has been fixed; sorry about that.

For some strange reason, I wasn't able to run `make` without errors on `master`, but it now seems to work after a full recompile. :slightly_smiling_face:

File: test/tuple.jl [hunk 2]
```diff
@@ -1,7 +1,50 @@
...
 
-@test convert(Tuple, (1,2)) == (1,2)
-@testset "indexing" begin
+struct BitPerm_19352
+    p::NTuple{8,UInt8}
+    function BitPerm(p::NTuple{8,UInt8})
+        sort(collect(p)) != collect(0:7) && error("$p is not a permutation of 0:7")
+        new(p)
+    end
+    BitPerm_19352(xs::Vararg{Any,8}) = BitPerm(map(UInt8, xs))
```
Comment (tkelman) [inline test/tuple.jl]: this is a bit of a refactoring of the constructor, hopefully it would still reproduce the original issue in this form?

Comment (HarrisonGrodin) [inline test/tuple.jl]: @tkelman Yep! I tested it in v0.5, where the bug still existed, and it still produces the same behavior. Just a bit more concise.

**Review by tkelman — APPROVED** [review]
looks like everything's still here despite a bit of reordering, so lgtm - should definitely be squashed on merging since intermediate commits failed

**Pull request merged**